application/src/test/java/com/sanction/lightning/resources/FacebookResourceTest.java

```diff
@@ -120,27 +126,92 @@ public void testGetMediaBytesWithNullUrl() {
+    when(urlService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
+    when(urlService.fetchInputStreamFromConnection(urlConnection)).thenReturn(inputStream);
+    when(urlService.inputStreamToByteArray(any(InputStream.class))).thenReturn(null);
+
     Response response = resource.getMediaBytes(key, "Test");
 
     assertEquals(response.getStatusInfo(), Response.Status.INTERNAL_SERVER_ERROR);
   }
 
   @Test
+  public void testPublishWithNullUsername() {
+    Response response = resource.publish(key, null, null, null, null);
+
+    assertEquals(response.getStatusInfo(), Response.Status.BAD_REQUEST);
+  }
+
+  @Test
+  public void testPublishWithNullMessage() {
+    Response response = resource.publish(key, "Test", null, null, null);
+
+    assertEquals(response.getStatusInfo(), Response.Status.BAD_REQUEST);
+  }
+
+  @Test
+  public void testPublishWIthNullInputStream() {
+    Response response = resource.publish(key, "Test", null, null, "Test");
+
+    assertEquals(response.getStatusInfo(), Response.Status.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void testPublishWithNullBytes() {
+    when(urlService.inputStreamToByteArray(inputStream)).thenReturn(null);
+    Response response = resource.publish(key, "Test", inputStream, contentDisposition, "Test");
+
+    assertEquals(response.getStatusInfo(), Response.Status.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void testPublishWithNullFacebookResponse() {
+    byte[] testBytes =  {};
+    when(urlService.inputStreamToByteArray(inputStream)).thenReturn(testBytes);
+    when(facebookService.publishToFacebook(any(byte[].class),
+            any(String.class), any(String.class))).thenReturn(null);
+    Response response = resource.publish(key, "Test", inputStream, contentDisposition, "Test");
+
+    assertEquals(response.getStatusInfo(), Response.Status.INTERNAL_SERVER_ERROR);
+  }
+
+  @Test
+  public void testPublish() {
+    byte[] testBytes =  {};
+    when(urlService.inputStreamToByteArray(inputStream)).thenReturn(testBytes);
+    when(facebookService.publishToFacebook(any(byte[].class),
+            any(String.class), any(String.class))).thenReturn(facebookPhoto);
+    Response response = resource.publish(key, "Test", inputStream, contentDisposition, "Test");
+
+    assertEquals(response.getStatusInfo(), Response.Status.OK);
+    assertEquals(response.getEntity(), facebookPhoto);
+  }
+
+  @Test
   public void testGetMediaBytes() {
-    byte[] testBytes = {};
-    when(urlDownloadService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
-    when(urlDownloadService.inputStreamToByteArray(any(URLConnection.class))).thenReturn(testBytes);
+    byte[] testBytes =  {};
+    when(urlService.fetchUrlConnection(any(String.class))).thenReturn(urlConnection);
+    when(urlService.fetchInputStreamFromConnection(urlConnection)).thenReturn(inputStream);
+    when(urlService.inputStreamToByteArray(any(InputStream.class))).thenReturn(testBytes);
```

Comment:
Same thing here with the inputStream, probably want to `when()` the actual object instead of `any(InputStream.class)`